src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const key in obj.extraData) {
+          stateCopy[key] = obj.extraData[key];
+        }
+        stateHistory.push({ ...stateCopy });
```

Comment:
You are pushing the state copy into the `stateHistory` array after each action. This operation is repeated in every case and can be moved out of the `switch` statement to follow the DRY (Don't Repeat Yourself) principle.

Reply:
fixed